util/loghttp/loghttp.go

```diff
@@ -37,6 +40,70 @@ func init() {
 			}
 		}
 		resp.Body = ioutil.NopCloser(buf)
-		log.Printf("<-- %d %s %s\n", resp.StatusCode, resp.Request.URL, string(buf.Bytes()))
+		log.Printf("Outgoing <-- %d %s %s\n", resp.StatusCode, resp.Request.URL, string(buf.Bytes()))
 	}
 }
+
+// Wrap wraps the given handler with a Handler that logs HTTP requests
+// and responses.
+func Wrap(handler http.Handler) Handler {
+	return Handler{wrapped: handler}
+}
+
+// Handler is a wrapper for http.Handlers that logs the HTTP request and
+// response. It logs full request headers but logging full response headers
+// seems like more work (eg
+// https://stackoverflow.com/questions/29319783/logging-responses-to-incoming-http-requests-inside-http-handlefunc)
+// so we settle for logging the response status code and response body for now.
+type Handler struct {
+	wrapped http.Handler
+}
+
+// ServeHTTP logs the request, calls the underlying handler, and logs the response.
+func (l Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	dump, err := httputil.DumpRequest(r, true)
```

Comment:
Should the outgoing request path also use `DumpRequest` to match?

Reply:
look like you already made this change :)